src/components/TimeSeriesCard/TimeSeriesCard.jsx

```diff
@@ -156,10 +156,10 @@ export const formatChartData = (timeDataSourceId = 'timestamp', series, values)
       // First filter based on on the dataFilter
       const filteredData = filter(values, dataFilter);
       if (!isEmpty(filteredData)) {
-        // have to filter out null values from the dataset, as it causes Carbon Charts to break
         filteredData
           .filter(dataItem => {
-            return !isNil(dataItem[dataSourceId]) && dataItem[timeDataSourceId] === timestamp;
+            // only allow valid timestamp matches
+            return !isNil(dataItem[timeDataSourceId]) && dataItem[timeDataSourceId] === timestamp;
```

Comment:
Isn't this still filtering out the null values here?

Reply:
This is filtering the null or undefined timestamps. Without this, we wouldn't know when the null value came in. The swap here was to stop filtering out the actual null data values. The timestamp should still be valid though